lib/xandra/protocol.ex

```diff
@@ -306,6 +306,12 @@ defmodule Xandra.Protocol do
     <<date::32>>
   end
 
+  defp encode_value(:decimal, float) when is_float(float) do
+    encode_value(:decimal, Decimal.new(float))
+  end
```

Comment:
Not sure about this casting since `:float` types exist as well. Is there casting when encoding other types?